lib/stream/leo-stream.js

```diff
@@ -635,10 +637,14 @@ module.exports = function(configure) {
 					logger.time("kinesis request");
 					kinesis.putRecords({
 						Records: records.map((r) => {
+							let randomhashkey = 0
+							if(!opts.partitionHashKey || opts.partitionHashKey == "random") {
+								randomhashkey = uuidv4()	
+							}
 							return {
 								Data: r,
 								PartitionKey: "0",
-								ExplicitHashKey: (opts.partitionHashKey || 0).toString()
+								ExplicitHashKey: (opts.partitionHashKey || randomhashkey).toString()
```

Comment:
@Handyman @mrn3 @burleyb 
Using a random partitions is problematic and will not guarantee that events will be received in the correct order.  To allow scaling and maintain ordering we need to set `PartitionKey` field to the target queue and set `ExplicitHashKey` to undefined